### PR TITLE
Add ability to set timeout on fetch and fetch-and-modify operations

### PR DESCRIPTION
**Pull request by tanya-coding**
This will set maxTime in milliseconds: https://mongodb.github.io/mongo-java-driver/3.4/javadoc/com/mongodb/DBCursor.html#maxTime-long-java.util.concurrent.TimeUnit-

### CODE REVIEW

File: src/somnium/congomongo.clj [hunk 1]
```diff
@@ -585,19 +589,23 @@ You should use fetch with :limit 1 instead."))); one? and sort should NEVER be c
        :return-new? -> if true, the updated document is returned,
                        otherwise the old document is returned
                        (or it would be lost forever)
-       :upsert?     -> do upsert (insert if document not present)"
+       :upsert?     -> do upsert (insert if document not present)
+       :max-time-ms -> set the maximum execution time for operations"
   {:arglists '([collection where update {:only nil :sort nil :remove? false
-                                         :return-new? false :upsert? false :from :clojure :as :clojure}])}
-  [coll where update & {:keys [only sort remove? return-new? upsert? from as]
+                                         :return-new? false :upsert? false :max-time-ms 0 :from :clojure :as :clojure}])}
```
Comment (glenjamin) [inline src/somnium/congomongo.clj]: this defaults to `nil` on fetch, but `0` here, is that intentional?

Comment (tanya-coding) [reply src/somnium/congomongo.clj]: Yes, due to differences in how we use it further. We use `(when...` and mutate cursor in fetch but fetch-and-modify just uses straight up Java method.

Comment (glenjamin) [reply src/somnium/congomongo.clj]: ah right, your other option would be a conditional and using different overloads, which isn't an improvement 👍

**Comment by glenjamin** [comment]
Github's hidden it because of the force-push, but see
https://github.com/circleci/congomongo/commit/c9a77be275c0460efc4c1f8b3eba4192029c878f
and
https://github.com/circleci/congomongo/commit/0b074fedb3723ef8f60bcf10519b2b898defe657

**Pull request merged**